internal/resources/metal/port/resource.go

```diff
@@ -127,20 +130,20 @@ func resourceMetalPort() *schema.Resource {
 
 func resourceMetalPortUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	start := time.Now()
-	cpr, _, err := getClientPortResource(d, meta)
+	cpr, _, err := getClientPortResource(ctx, d, meta)
 	if err != nil {
 		return diag.FromErr(equinix_errors.FriendlyError(err))
 	}
 
 	for _, f := range [](func(*ClientPortResource) error){
 		portSanityChecks,
```

Comment:
I'm tempted to ask for a contextDisposal wrapper so that this one function can accept and noop a supplied context.

The other functions wouldn't need to be created with a context and context would be supplied at time of use.

something like:
```go
func contextDisposal(f func(*ClientPortResource) error) func(_ context.Context, *ClientPortResource) error){
  return func(_ context.Context, *ClientPortResource) error) {
    return f
  }
}
```
```go
	for _, f := range [](func(context.Context, *ClientPortResource) error){
		contextDisposal(portSanityChecks),
... (the other lines would not change, so function names without arguments)
	} {
		if err := f(ctx, cpr); err != nil {
```

Reply:
A wrapper function wouldn't be sufficient here because the existing function bodies need to be updated to reference the context.  Instead I updated the various helper functions to accept a context and a ClientPortResource so that the PR diff is easier to parse.